include/seqan3/std/span

```diff
@@ -70,12 +70,26 @@ inline constexpr bool enable_safe_range;
 template<class span_tp>
     requires std::is_span<span_tp>::value
 inline constexpr bool enable_safe_range<span_tp> = true;
+
+// ranges library defines enable_view as:
+// https://github.com/ericniebler/range-v3/blob/6103268542c27210dbc3f99f1fa0c1e348b52184/include/range/v3/range/concepts.hpp#L201
+//
+// whereas we allow const and reference versions.
+//
+// the standard says:
+//   Pursuant to [namespace.std], users may specialize enable_­view to true for cv-unqualified program-defined types
+//   which model view, and false for types which do not. Such specializations shall be usable in constant expressions
+//   ([expr.const]) and have type const bool.
+// https://eel.is/c++draft/range.view#5
+template<class span_tp>
+    requires std::is_span<span_tp>::value
+inline constexpr bool enable_view<span_tp> = true;
```

Comment:
This is still valid? 

> #if RANGE_V3_VERSION >= 901

Reply:
yeah this is still valid. the above `#if RANGE_V3_VERSION < 1000` could be taken out, because we require at least `>= 1000`. 

https://github.com/seqan/seqan3/blob/93de9c772d9cf96966439a525e4907eea56414c5/include/seqan3/core/platform.hpp#L69-L87

Reply:
Right. But it also means we could change the entire condition to only allow `RANGE_V3_VERSION >= 1000` or not?

Reply:
done